angular/projects/spark-angular/src/lib/directives/sprk-spinner/sprk-spinner.directive.ts

```diff
@@ -0,0 +1,33 @@
+import { AfterViewInit } from '@angular/core';
+import { Input } from '@angular/core';
+import { Directive, ElementRef, Renderer2 } from '@angular/core';
+
+@Directive({
+  selector: '[sprkSpinner]',
+})
+export class SprkSpinnerDirective implements AfterViewInit {
+  /**
+   * @ignore
+   */
+  constructor(private renderer: Renderer2, private el: ElementRef) {}
+
```

Comment:
This needs to have accessibility baked in too. Currently for the spinner in button we have an input for aria-label set to 'Loading' and then they can override it. We'll want to have that here. But instead of it being based on isSpinning. It'll always be there because this directive being on an HTML element means it is spinning.

```
  /**
   * Optional string value that is
   * set for the `aria-label`
   * when `isSpinning` is `true`.
   */
  @Input() spinningAriaLabel = 'Loading';
```

Reply:
from what I can understand, the ariaLabel is being applied to the button.
Line 129 of the second image shows that the el is referencing the actual button, and not the spinner.

From a cursary search, aria-label isn't even the best way of handling a button that loads.
I created an issue: https://github.com/sparkdesignsystem/Backlog/issues/1298

Do we want to continue with encouraging people to use it?

![image](https://user-images.githubusercontent.com/4342363/96033137-b4871180-0e2d-11eb-900c-e0df77ee06f0.png)
![image](https://user-images.githubusercontent.com/4342363/96033225-d1234980-0e2d-11eb-9724-ee5b01967edf.png)
![image](https://user-images.githubusercontent.com/4342363/96033264-de403880-0e2d-11eb-9f0d-85bc3157f18e.png)

Reply:
Well in this compositional way, I think we could allow the implementor to add the relevant tags if they turn the button into a loading state.

We could add it into the spinning stories and dev docs
Of course, the thing is this is a different research task that I made an issue for.


Here's my take:
- Allow the Button directive to apply ariaLabel (set by the user, and is already happening in production)
- that ariaLabel doesn't seem like the best option anyways, but it at least is consistent until we can work the research on this story (https://github.com/sparkdesignsystem/Backlog/issues/1298)
- The spinner does not have any aria roles on it.

Reply:
The button's aria-label solution was only in place because the spinner was a part of buttons, but like you said, its not the right solution. I think you should do the research around the spinner's a11y as part of this task. I don't like the idea of shipping an element that represents meaningful state and is invisible to screen readers.